csrc/multidevice/executor.cpp

```diff
@@ -262,7 +260,7 @@ std::ostream& MultiDeviceExecutor::print() {
       communication_counter++;
     } else {
       debug() << "Compute segment " << compute_segment_counter << ":{\n";
-      auto fusion = staged_fusion_->makeFusion(group);
+      auto&& [cloner, fusion] = staged_fusion_->makeFusion(group);
```

Comment:
nitpick: I see a few places where we are merely trying to print the group.

`makeFusion` clones the fusion, while if we do `group->getFusion()`, we are going to be using a cached one instead?
My question is, is every `group` supposed to have a cached fusion at this/some point and will we be saving some fusion copy then?